src/main/java/it/gov/pagopa/rtd/ms/rtdmsdecrypter/service/DecrypterImpl.java

```diff
@@ -79,17 +79,20 @@ public BlobApplicationAware decrypt(BlobApplicationAware blob) {
       blob.setStatus(BlobApplicationAware.Status.DECRYPTED);
       log.info("Blob {} decrypted.", blob.getBlob());
 
-    } catch (Exception ex) {
-      // Should throw an IOException just like decryptFile, this creates problems in the event
-      // handler where the method chaining doesn't allow Exception throws
-      log.error("Cannot decrypt {}: {}", blob.getBlob(), ex.getMessage());
+    } catch (IOException e) {
+      log.warn("{}: {}", e.getMessage(), blob.getBlob());
+    } catch (PGPException e) {
+      log.error("Cannot decrypt {}: {}", blob.getBlob(), e.getMessage());
+    } catch (IllegalArgumentException e) {
+      log.error("Cannot decrypt {}: {}", blob.getBlob(), e.getMessage());
     }
 
     return blob;
   }
 
   @SneakyThrows
```

Comment:
Given that the method is now declaring checked exceptions I think the annotation @SneakyThrows should be removed.

Reply:
I agree with you, I'll fix it.
I had to specify the thrown exception in order to catch them in decrypt method.
This grants a granular control over errors.
The SneakyThrows annotation is now useless.